src/qibolab/pulses.py

```diff
@@ -644,359 +651,62 @@ class Pulse:
-                self._duration.value = int(value)
-            elif isinstance(value, int):
-                self._duration.value = value
-        else:
-            if isinstance(value, np.integer):
-                self._duration = int(value)
-            else:
-                self._duration = value
-
-        if not self._start is None:
-            if (
-                isinstance(self._start, se_int)
-                or isinstance(self._duration, se_int)
-                or isinstance(self._finish, se_int)
-            ):
-                self._finish = se_int(self._start + self._duration)["_p" + str(self._id) + "_finish"]
-            else:
-                self._finish = self._start + self._duration
-
-    @property
-    def finish(self) -> int:
-        """Returns the time when the pulse is scheduled to finish.
-
-        Calculated as pulse.start - pulse finish.
-        """
-        if isinstance(self._finish, se_int):
-            return self._finish.value
-        return self._finish
-
-    @property
-    def se_start(self) -> se_int:
-        """Returns a symbolic expression for the pulse start."""
-
-        if not isinstance(self._start, se_int):
-            self._start = se_int(self._start)["_p" + str(self._id) + "_start"]
-        return self._start
-
-    @property
-    def se_duration(self) -> se_int:
-        """Returns a symbolic expression for the pulse duration."""
-
-        if not isinstance(self._duration, se_int):
-            self._duration = se_int(self._duration)["_p" + str(self._id) + "_duration"]
-        return self._duration
-
-    @property
-    def se_finish(self) -> se_int:
-        """Returns a symbolic expression for the pulse finish."""
+    start: Optional[int] = None
+    """Time when the pulse is scheduled to be played, in ns."""
+    duration: Optional[int] = None
+    """Duration of the pulse, in ns."""
+    amplitude: Optional[float] = None
+    """Pulse amplitude.
 
-        if not isinstance(self._finish, se_int):
-            self._finish = se_int(self._finish)["_p" + str(self._id) + "_finish"]
-        return self._finish
+    Pulse amplitudes are normalised between -1 and 1.
+    """
+    frequency: Optional[int] = None
+    """Frequency of the pulse, in Hz."""
+    relative_phase: Optional[float] = None
+    """Relative phase of the pulse, in radians."""
+    shape: Optional[PulseShape] = None
+    """Pulse shape, as a PulseShape object."""
+    channel: int =0,
+    """Channel on which the pulse should be played.
+
+    When a sequence of pulses is sent to the platform for execution,
+    each pulse is sent to the instrument responsible for playing pulses
+    the pulse channel. The connection of instruments with channels is
+    defined in the platform runcard.
+    """
+    type: PulseType =PulseType.DRIVE,
+    """Pulse type, as an element of PulseType enumeration."""
+    qubit: Optional[int] =0,
+    """Qubit addressed by the pulse."""
+    if: int = 0
```

Comment:
:scream: :scream: :scream:

Reply:
I know, but I was lacking a better name. I can append a `_`, but I'd rather choose a more explicit name.

Suggestions?

Reply:
You can just be redundant and put `intermediate_frequency` or `intermediate_freq` if you want. I know that it is super boring but I believe that it does the job better than `if`.

Reply:
It just looks funny with the editor coloring and the `:`.

I think here `if` is short for "intermediate frequency". I think it was `_if` before.

It's a bit weird that we need this in the pulse, because it is generally instrument dependent (`=pulse.frequency - local_oscillator.frequency`), so maybe we will even drop it at some point. I would keep it for the first round, as I am not sure where it is used, but I wouldn't spend much time on it.

Reply:
> I think here `if` is short for "intermediate frequency". I think it was `_if` before.

I know, and it was.
I'm tempted to use `intermediate_frequency`, as @andrea-pasquale said.

> It's a bit weird that we need this in the pulse, because it is generally instrument dependent (`=pulse.frequency - local_oscillator.frequency`), so maybe we will even drop it at some point. I would keep it for the first round, as I am not sure where it is used, but I wouldn't spend much time on it.

I was also tempted to drop, since it is quite decoupled from everything else. But I agree to keep changes minimal.
Maybe I can just keep `_if`, and just touch as little as possible (until we decide to drop it for good).